### PR TITLE
feat(Styles): Create system styles and gamut theme

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview

Partial Rebase of https://github.com/Codecademy/client-modules/pull/953

### Changes
- Reformats typography variables to be more friendly and single-level depth.  (fontSize, fontWeight, etc).  This includes some small updates to existing documentation.
- Adds more intervals on spacing (more multiples at 4 at lower levels).
- Exports and initial `theme` object for use with emotion (this is only a subset right now and will be added to) and configures the emotion definition.
- Adds our theme to storybook previews to ensure stories behave correctly.
- Adds `gamut-system` and creates our initial set of typed style functions for future use.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-styles/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+import { Theme as GamutTheme } from './theme';
+
 export * from './cache';
 export * from './variables';
 export * from './utilities';
+export * from './system';
+export * from './theme';
+
+declare module '@emotion/react' {
+  /* eslint-disable-next-line @typescript-eslint/no-empty-interface */
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/index.ts:11]: We should probably just remove that lint rule...

File: packages/gamut-styles/src/variables/spacing.ts [hunk 2]
```diff
@@ -6,11 +6,16 @@ export const baseUnit = pxRem(base);
...
+  0: 0,
   4: pxRem(4),
   8: pxRem(8),
+  12: pxRem(12),
   16: pxRem(16),
+  20: pxRem(20),
   24: pxRem(24),
+  28: pxRem(28),
   32: pxRem(32),
+  36: pxRem(36),
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/variables/spacing.ts:18]: I feel like it kinda defeats the purpose of having solidified spacing constants if we have them at every 4... can we keep to what we already have?

(same with font size below)

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by jakemhiller — APPROVED** [review]

File: packages/gamut-styles/src/variables/spacing.ts (hunk 2 above)
Comment (jakemhiller) [reply packages/gamut-styles/src/variables/spacing.ts:18]: i assume this was done based on need in designs?

Comment (JoshuaKGoldberg) [reply packages/gamut-styles/src/variables/spacing.ts:18]: It feels like we're starting to see _two_ kinds of spacing constant requests:

* Important common ones which already existed here and are used as the primary spacing measurements in designs
* Random other measurements that happen to align to 4 and are necessary once in a while for design tweaks

...maybe we should have two groups for them, like with colors?

Comment (codecaaron) [reply packages/gamut-styles/src/variables/spacing.ts:18]: I think there is probably a need for 4, 8, 12, 16 to be at the 4 intervals.  But larger spaces get unnecessary.  Pruning these for now.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.0.1-alpha.78f3af.0<br/>@codecademy/gamut-styles@7.2.1-alpha.78f3af.0<br/>@codecademy/gamut-system@0.1.0-alpha.78f3af.0<br/>@codecademy/gamut@17.0.1-alpha.78f3af.0<br/>@codecademy/markdown-overrides@0.3.7-alpha.78f3af.0<br/>@codecademy/styleguide@18.0.2-alpha.78f3af.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa961de2629f4dc8c4d5448--gamut-preview.netlify.app>https://5fa961de2629f4dc8c4d5448--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa961de2629f4dc8c4d5448>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.0.2-alpha.21d412.0<br/>@codecademy/gamut-styles@7.2.1-alpha.21d412.0<br/>@codecademy/gamut@17.1.1-alpha.21d412.0<br/>@codecademy/markdown-overrides@0.3.7-alpha.21d412.0<br/>@codecademy/styleguide@18.1.1-alpha.21d412.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa99a15f7c21e0096104352--gamut-preview.netlify.app>https://5fa99a15f7c21e0096104352--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa99a15f7c21e0096104352>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.0.2-alpha.63cfb4.0<br/>@codecademy/gamut-styles@7.2.1-alpha.63cfb4.0<br/>@codecademy/gamut@17.1.1-alpha.63cfb4.0<br/>@codecademy/markdown-overrides@0.3.7-alpha.63cfb4.0<br/>@codecademy/styleguide@18.1.2-alpha.63cfb4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5faa0f1c938db700a1733d04--gamut-preview.netlify.app>https://5faa0f1c938db700a1733d04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5faa0f1c938db700a1733d04>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5faabb59c1fe6df246434ce4--gamut-preview.netlify.app>https://5faabb59c1fe6df246434ce4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5faabb59c1fe6df246434ce4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.0.2-alpha.6875a9.0<br/>@codecademy/gamut-styles@7.2.1-alpha.6875a9.0<br/>@codecademy/gamut@17.1.1-alpha.6875a9.0<br/>@codecademy/markdown-overrides@0.3.7-alpha.6875a9.0<br/>@codecademy/styleguide@18.1.3-alpha.6875a9.0<br/></pre>

**Pull request merged**